modules/tensor_mechanics/examples/piston/piston_params.i

```diff
@@ -0,0 +1,152 @@
+## This example is documented on YouTube at:
+## https://www.youtube.com/watch?v=L9plLYkAbvQ
+## 
+## Additional files (e.g. the CAD model, results)
+## can be downloaded freely from Zenodo at:
+## https://doi.org/10.5281/zenodo.3886965
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Mesh]
+  type = FileMesh #Read in mesh from file
+  file = piston_coarse.e
+[]
+
+# This is where mesh adaptivity magic happens:
+[./Adaptivity]
+  steps = 1
+  #initial_steps = 2
+  max_h_level = 3
+  cycles_per_step = 1
+  initial_marker = uniform
+  marker = errorFraction
+  [./Markers]
+    [./uniform]
+      type = UniformMarker
+      mark = refine
+    [../]
+    
+    [./errorFraction]
+      type = ErrorFractionMarker
+      coarsen = 0.5
+      indicator = gradientJump
+      refine = 0.5
+    [] 
+ [../]
+
+  [./Indicators]
+    [./gradientJump]
+      type = GradientJumpIndicator
+      variable = disp_y
+    []
+  []
+[../]
+
+[Variables]
+  [./disp_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
```

Comment:
Since you use `add_variables = true` in Master Action, you do not need to list those variables.

Reply:
See https://github.com/idaholab/moose/pull/18543/commits/5696be02cd2e34ed99a48da6963927b6f263bb42